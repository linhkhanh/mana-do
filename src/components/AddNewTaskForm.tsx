import React, { useReducer, useRef } from 'react';
import { useHistory } from 'react-router-dom';
import Service from '../service';
import { createTodo } from '../store/actions';
import reducer, { initialState } from '../store/reducer';

const AddNewTaskForm = (props: any) => {
    const inputRef = useRef<HTMLInputElement>(null);
    const [{ todos }, dispatch] = useReducer(reducer, initialState);
    const history = useHistory();
    const getNewTodo = props.getNewTodo;

    const onCreateTodo = async (e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.key === 'Enter' && inputRef.current) {
            try {
                const resp = await Service.createTodo(inputRef.current.value);
                dispatch(createTodo(resp));
                getNewTodo(resp);
                inputRef.current.value = '';
            } catch (e) {
                if (e.response.status === 401) {
                    history.push('/')
                }
            }
        }
    }
    return (

        <div className="Todo__creation floating-label-group">
            <input
                ref={inputRef}
                className="Todo__input"
                onKeyDown={onCreateTodo}
                required
            />
            <label className="floating-label-1">What need to be done?</label>
        </div>
    )
}

export default AddNewTaskForm