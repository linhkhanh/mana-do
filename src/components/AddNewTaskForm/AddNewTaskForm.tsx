import React from 'react';

const AddNewTaskForm = (props: any) => {
    const inputRef = props.inputRef;
    const onCreateTodo = props.onCreateTodo;

    return (
        <div className="Todo__creation floating-label-group">
            <input
                ref={inputRef}
                className="Todo__input"
                onKeyDown={onCreateTodo}
                required
            />
            <label className="floating-label-1">What need to be done?</label>
        </div>
    )
}

export default AddNewTaskForm