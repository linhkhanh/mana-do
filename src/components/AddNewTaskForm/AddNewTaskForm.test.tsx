import Enzyme from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import AddNewTaskForm from './AddNewTaskForm';
import { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });

it('should render properly', () => {
    const tree = renderer.create(<AddNewTaskForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render an input tag', () => {
    const wrapper = shallow(<AddNewTaskForm />);
    expect(wrapper.find('input').exists()).toBe(true);
  });
