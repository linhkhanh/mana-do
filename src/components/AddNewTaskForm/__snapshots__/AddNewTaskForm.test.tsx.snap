// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<div
  className="Todo__creation floating-label-group"
>
  <input
    className="Todo__input"
    required={true}
  />
  <label
    className="floating-label-1"
  >
    What need to be done?
  </label>
</div>
`;
